main.js

```diff
@@ -379,40 +377,52 @@ function addWidgetsFromJSON(jsonObject, numberOfColumns, numberOfRows) {
         const row = Math.floor(i / numberOfColumns) + 1;
         const key = column + "_" + row;
         const widget = widgetsDictionary[key];
-        if (!widget || existingWidgets.indexOf(widget) >= 0) {
-            //no widget found for this part of the grid, add an empty cell instead
-            if (!widget && existingWidgets.indexOf(widget) < 0) {
-                const id = addCell();
-                makeHiddenCell(document.getElementById(id));
-                // console.log("HIDDEN KEY : " + key);
+
+        if (widget == undefined) {
+            // widget is not in the available list, hide
+            const id = addCell();
+            makeHiddenCell(document.getElementById("widget-box-" + id));
+        } else {
+            if (jsonObject.config[widget.config].widgetTitle !== undefined) {
```

Comment:
If there's any chance that `jsonObject.config[widget.config]` will not exist here, you'll probably want to add a check for that first and `&&` it with this current check.

Reply:
Yep